ctru-rs/src/services/cam.rs

```diff
@@ -0,0 +1,1050 @@
+            let r = ctru_sys::CAMU_ReadMcuVariableI2cExclusive(&mut data, camera.bits(), addr);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(data)
+            }
+        }
+    }
+
+    pub fn set_image_quality_calibration_data(
+        &self,
+        data: ImageQualityCalibrationData,
+    ) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_SetImageQualityCalibrationData(data.0);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn get_image_quality_calibration_data(&self) -> crate::Result<ImageQualityCalibrationData> {
+        unsafe {
+            let mut data = ImageQualityCalibrationData::default();
+            let r = ctru_sys::CAMU_GetImageQualityCalibrationData(&mut data.0);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(data)
+            }
+        }
+    }
+
+    pub fn set_package_parameter_without_context(
+        &self,
+        param: PackageParameterCameraSelect,
+    ) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_SetPackageParameterWithoutContext(param.0);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn set_package_parameter_with_context(
+        &self,
+        param: PackageParameterContext,
+    ) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_SetPackageParameterWithContext(param.0);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn set_package_parameter_with_context_detail(
+        &self,
+        param: PackageParameterContextDetail,
+    ) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_SetPackageParameterWithContextDetail(param.0);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn play_shutter_sound(&self, sound: CamShutterSoundType) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_PlayShutterSound(sound.bits());
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn driver_initialize(&self) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_DriverInitialize();
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn driver_finalize(&self) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_DriverFinalize();
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
```

Comment:
Is it safe to call these functions multiple times? Is `driver_initialize` required for any of the other functions? If either answers are yes, maybe we should have another struct to enforce the init/deinit and function access.

Edit: looking at the docs for `camInit` and `camExit`, they call these functions automatically. Maybe we shouldn't expose them then?

Reply:
Same idea for this as the register functions